comtypes/tools/codegenerator.py

```diff
@@ -186,6 +186,7 @@ def __init__(self, known_symbols=None):
         self.stream = io.StringIO()
         self.imports = ImportedNamespaces()
         self.declarations = DeclaredNamespaces()
+        self.type_name = TypeNamer().type_name
```

Comment:
It should be `TypeNamer().type_name()` or the method should be defined as property.

Reply:
I don't feel right about methods and properties returning `Callable` objects.

Instead, how about making the `TypeNamer` itself a `Callable` object?

https://github.com/enthought/comtypes/blob/4f3a009c770c6cb1979b910325a8716153fb5bb5/comtypes/tools/codegenerator.py#L1134-L1136
↓
```py
class TypeNamer(object):
    def __call__(self, t):
        # type: (...) -> str
        ...
```

https://github.com/enthought/comtypes/blob/4f3a009c770c6cb1979b910325a8716153fb5bb5/comtypes/tools/codegenerator.py#L183-L189
↓
```py
class CodeGenerator(object):

    def __init__(self, known_symbols=None):
        self.stream = io.StringIO()
        self.imports = ImportedNamespaces()
        self.declarations = DeclaredNamespaces()
        self.type_name = TypeNamer()
        ...
```

Reply:
> I don't feel right about methods and properties returning `Callable` objects.

In other words, it is difficult for me to give proper names to methods and properties that return functions.

`type_name` seems to be returning a string, not a function.

I think it should be `type_name_func` in property case (or `get_type_name_func` in method case).
But their names are ugly for me.

So I first tried to assign the instance method itself.